sdk/cognitiveservices/textanalytics/Azure.AI.TextAnalytics/tests/TextAnalyticsClientTests.cs

```diff
@@ -0,0 +1,177 @@
+                {
+                    new NamedEntity("EntityText0", "EntityType0", "EntitySubType0", 0, 1, 0.5),
+                    new NamedEntity("EntityText1", "EntityType1", "EntitySubType1", 0, 1, 0.5),
+                }),
+                new RecognizeEntitiesResult("2", new TextDocumentStatistics(), new List<NamedEntity>()
+                {
+                    new NamedEntity("EntityText0", "EntityType0", "EntitySubType0", 0, 1, 0.5),
+                    new NamedEntity("EntityText1", "EntityType1", "EntitySubType1", 0, 1, 0.5),
+                }),
+            };
+            var mockResultCollection = new RecognizeEntitiesResultCollection(mockResults,
+                new TextBatchStatistics(2, 2, 0, 2),
+                "modelVersion");
+
+            var mockResponse = new MockResponse(200);
+            mockResponse.SetContent(SerializationHelpers.Serialize(mockResultCollection, SerializeRecognizeEntitiesResultCollection));
+
+            var mockTransport = new MockTransport(mockResponse);
+            TextAnalyticsClient client = CreateTestClient(mockTransport);
+
+            var inputs = new List<TextDocumentInput>()
+            {
+                new TextDocumentInput("1", "TextDocument1"),
+                new TextDocumentInput("2", "TextDocument2"),
+            };
+
+            var response = await client.RecognizeEntitiesAsync(inputs, new TextAnalysisOptions());
+            var resultCollection = response.Value;
+
+            Assert.AreEqual("1", resultCollection[0].Id);
+            Assert.AreEqual("2", resultCollection[1].Id);
+        }
+
+        [Test]
+        public async Task RecognizeEntitiesResultsSorted_WithErrors()
+        {
+            var mockResults = new List<RecognizeEntitiesResult>()
+            {
+                new RecognizeEntitiesResult("2", new TextDocumentStatistics(), new List<NamedEntity>()
+                {
+                    new NamedEntity("EntityText0", "EntityType0", "EntitySubType0", 0, 1, 0.5),
+                    new NamedEntity("EntityText1", "EntityType1", "EntitySubType1", 0, 1, 0.5),
+                }),
+                new RecognizeEntitiesResult("3", new TextDocumentStatistics(), new List<NamedEntity>()
+                {
+                    new NamedEntity("EntityText0", "EntityType0", "EntitySubType0", 0, 1, 0.5),
+                    new NamedEntity("EntityText1", "EntityType1", "EntitySubType1", 0, 1, 0.5),
+                }),
+                new RecognizeEntitiesResult("4", "Document is invalid."),
+                new RecognizeEntitiesResult("5", "Document is invalid."),
+            };
+            var mockResultCollection = new RecognizeEntitiesResultCollection(mockResults,
+                new TextBatchStatistics(2, 2, 2, 2),
+                "modelVersion");
+
+            var mockResponse = new MockResponse(200);
+            mockResponse.SetContent(SerializationHelpers.Serialize(mockResultCollection, SerializeRecognizeEntitiesResultCollection));
```

Comment:
It looks like it would be easier to read if you just supply json as a string

Reply:
I'll think about this as I increase test coverage.  Thanks!

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/9091